include/zephyr/net/ethernet.h

```diff
@@ -988,7 +993,7 @@ int net_eth_get_hw_config(struct net_if *iface, enum ethernet_config_type type,
  *
  * @return 0 if ok, <0 if error
  */
-#if defined(CONFIG_NET_VLAN)
+#if defined(CONFIG_NET_VLAN) && NET_VLAN_MAX_COUNT > 0
```

Comment:
I wonder if this should be reduced to simply:

```suggestion
#if NET_VLAN_MAX_COUNT > 0
```

Reply:
I suppose that could be done too. The values are linked together although if not careful by the developer, the `NET_VLAN_MAX_COUNT` could be set to a value >0 even if `CONFIG_NET_VLAN` is not set (yes, this is an error and would probably not happen in practice).
I kind of prefer this as it is now easy to understand what the check means, so VLAN needs to be enabled and the VLAN count > 0.